mingw_conf.sh

```diff
@@ -17,60 +29,79 @@ MINGW_DLLTOOL="${MINGW_BASE}-dlltool"
 MINGW_WINDRES="${MINGW_BASE}-windres"
 MINGW_STRIP="${MINGW_BASE}-strip"
 MINGW_OBJCOPY="${MINGW_BASE}-objcopy"
-# -Distribution specific mingw settings-
-case "$OS" in
-    ubuntu)
-        mingwPosix="-posix"
-        mingwLibDir="/usr/lib/gcc"
-        mingwVersion="$(${MINGW_CPP}${mingwPosix} -dumpversion)"
-        mingwTargetLibDir="${mingwLibDir}/${MINGW_BASE}/${mingwVersion}"
-        mingwLibpthreadDir="/usr/${MINGW_BASE}/lib"
-        PTW32Include=/usr/share/mingw-w64/include
-        PTW32Lib=/usr/x86_64-w64-mingw32/lib
-       ;;
-    rhel)
-        mingwPosix=""
-        mingwLibDir="/usr/lib64/gcc"
-        mingwVersion="$(${MINGW_CPP}${mingwPosix} -dumpversion)"
-        mingwTargetLibDir="/usr/${MINGW_BASE}/sys-root/mingw/bin"
-        mingwLibpthreadDir="$mingwTargetLibDir"
-        PTW32Include=/usr/x86_64-w64-mingw32/sys-root/mingw/include
-        PTW32Lib=/usr/x86_64-w64-mingw32/sys-root/mingw/lib
-       ;;
-    suse)
-        mingwPosix=""
-        mingwLibDir="/usr/lib64/gcc"
-        mingwVersion="$(${MINGW_CPP}${mingwPosix} -dumpversion)"
-        mingwTargetLibDir="/usr/${MINGW_BASE}/sys-root/mingw/bin"
-        mingwLibpthreadDir="$mingwTargetLibDir"
-        PTW32Include=/usr/x86_64-w64-mingw32/sys-root/mingw/include
-        PTW32Lib=/usr/x86_64-w64-mingw32/sys-root/mingw/lib
-        ;;
-    *)
-        echo "$ID is unknown, automatic mingw configuration is not possible."
-        exit 1
-        ;;
-esac
+
+if [[ -n $USE_MINGW_LLVM ]]; then
+    # This package isn't currently provided by Linux distributions, we're
+    # fetching it from Github.
+    export PATH="$MINGW_LLVM_DIR/bin:$PATH"
```

Comment:
Mucking with `PATH` like this isn't nice.  What is relying on this export (it's not there for GCC)?

Reply:
mingw-gcc is installed using the distro package manager (e.g. apt), usually ending up in ``/usr/bin``. On the other hand, we're extracting mingw-llvm to the ``build.deps`` subdir.

Reply:
But a bunch other paths are set, including a "root" path.  Can't an absolute path be used there?  If not, it's not a blocker, just asking ;)